pkgs/development/python-modules/randomgen/default.nix

```diff
@@ -0,0 +1,46 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, cython
+, numpy
+, packaging
+, setuptools
+, setuptools-scm
+, wheel
+}:
+
+buildPythonPackage rec {
+  pname = "randomgen";
+  version = "1.23.1";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-gmMZ6Pz0YppGCvDIZJ32XFXYi8OEzN5B90hekqKuwXw=";
+  };
+
+  nativeBuildInputs = [
+    cython
+    numpy
+    packaging
+    setuptools
+    setuptools-scm
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    cython
+    numpy
+    setuptools
+    wheel
+  ];
+
+  pythonImportsCheck = [ "randomgen" ];
+
+  meta = with lib; {
+    description = "Random generator supporting multiple PRNGs";
+    homepage = "https://pypi.org/project/randomgen";
+    license = licenses.ncsa;
+    maintainers = with maintainers; [ ];
```

Comment:
Ideally, the package would have a maintainer (can be registered in https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix) who can be contacted in case of PRs or issues regarding the packaging.